File: stickler_theif.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;
typedef long long int ll;

// } Driver Code Ends
class Solution
{
    public:
    //Function to find the maximum money the thief can get.
    int FindMaxSum(int arr[], int n)
    {
        // Your code here
        if (n == 0) return 0;
        if (n == 1) return arr[0];
        int dp[n];
        dp[0] = arr[0];
        dp[1] = arr[1];
        for(int i = 2; i < n; ++i) {
            dp[i] = dp[i-2] + arr[i];
            if (i - 3 >= 0) {
                dp[i] = max(dp[i], dp[i-3] + arr[i]);
            }
        }
        return max(dp[n-1], dp[n-2]);
    }
};

//{ Driver Code Starts.
int main()
{
    //taking total testcases
	int t;
	cin>>t;
	while(t--)
	{
	    //taking number of houses
		int n;
		cin>>n;
		int a[n];
		
		//inserting money of each house in the array
		for(int i=0;i<n;++i)
			cin>>a[i];
		Solution ob;
		//calling function FindMaxSum()
		cout<<ob.FindMaxSum(a,n)<<endl;
	}
	return 0;
}

// } Driver Code Ends

File: dp/stickler_theif.py
# User function Template for python3


class Solution:
    # Function to find the maximum money the thief can get.
    def FindMaxSum(self, a, n):
        # code here
        if n == 1:
            return a[0]
        dp = [0] * n
        dp[0] = a[0]
        for i in range(1, n):
            dp[i] = a[i]
            if i - 3 >= 0:
                dp[i] += max(dp[i - 3], dp[i - 2])
            elif i - 2 >= 0:
                dp[i] += dp[i - 2]

        return max(dp[n - 1], dp[n - 2])


# {
# Driver Code Starts
# Initial Template for Python 3
import atexit
import io
import sys

sys.setrecursionlimit(10**6)
# Contributed by : Nagendra Jha

if __name__ == "__main__":
    test_cases = int(input())
    for cases in range(test_cases):
        n = int(input())
        a = list(map(int, input().strip().split()))
        ob = Solution()
        print(ob.FindMaxSum(a, n))
# } Driver Code Ends
